core/src/main/java/io/grpc/Deadline.java

```diff
@@ -139,4 +139,9 @@ public long timeRemaining(TimeUnit unit) {
     Preconditions.checkNotNull(scheduler, "scheduler");
     return scheduler.schedule(task, deadlineNanos - System.nanoTime(), TimeUnit.NANOSECONDS);
   }
+
+  @Override
+  public String toString() {
+    return String.valueOf(deadlineNanos);
```

Comment:
`Deadline.toString()` for sure is useful. But what should it look like? The problem I see with this implementation is, if someone logs a `Deadline` object, only deadlines from the same JVM process can be compared meaningfully, as the `System.nanoTime()` part of `deadlineNanos` is kinda random. 

Should I implement this to return a proper date/time instead?